core/trino-main/src/test/java/io/trino/sql/planner/planprinter/TestJsonRepresentation.java

```diff
@@ -60,6 +63,101 @@ public void setUp()
         queryRunner.createCatalog(TEST_SESSION.getCatalog().get(), new TpchConnectorFactory(1), ImmutableMap.of());
     }
 
+    @Test
+    public void testLogicalJsonPlan()
+    {
+        MaterializedResult actualPlan = queryRunner.execute("EXPLAIN (TYPE LOGICAL, FORMAT JSON) SELECT quantity FROM lineitem limit 10");
+        MaterializedResult expectedPlan = resultBuilder(queryRunner.getDefaultSession(), createVarcharType(2012))
+                .row("{\n" +
```

Comment:
Can we serialise the JSON back into JsonRenderedNode and assert equality on that ?
Or if you really want to assert on json string, then we should create expected json file in test/resources and use that instead of long string here.